### ISSUE TITLE
Blazor WASM MSAL Published Application Gets Stuck on "Completing Login" 

### ISSUE THREAD

**Issue by snapfisher** [dotnet/aspnetcore#52619]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Under .Net 8, a published Blazor WASM gets stuck in the login process, at "Completing Login...".  This occurs on Android Edge Chrome and Google Chrome browsers.  For some reason, the error does not occur on the Windows Edge browser.  This was a .Net 7 application which was converted to .Net 8.  This behavior did not occur in .Net 7.

To make it even stranger, about 1/3 of the time it does complete.  I have not been able to discern any pattern in this....and I have tried.

### Expected Behavior

Login should complete.  The user is correct.  The credentials are correct.  The service principal is set for SPA redirect URLs.  The callback URL is correctly entered.  In the app the client ID and Authority are correct.  (I tested with the tenant Guid and Name).   I originally had the SP set to supply Access and ID tokens.  I tried like this and, following the newish security guidance, removed both checkboxes.  There was no effect; the same problem occurred

### Steps To Reproduce

repo: snapfisher/french-vocab-blazor-bug

Note that I have seen: https://github.com/dotnet/aspnetcore/issues/51870  I do not know if it is the same issue, but the temp fix there had no effect on this.

### Exceptions (if any)

No exceptions.  Login does not complete.

### .NET Version

8.0.100

### Anything else?

IDE:  Visual Studio 2022, 17.8.3

PS S:\repos\frenchvocabblazorbug> dotnet --info
.NET SDK:
Version:           8.0.100
Commit:            57efcf1350
Workload version:  8.0.100-manifests.8d38d0cc

Runtime Environment:
OS Name:     Windows
OS Version:  10.0.22000
OS Platform: Windows
RID:         win-x64
Base Path:   C:\Program Files\dotnet\sdk\8.0.100\

.NET workloads installed:
Workload version: 8.0.100-manifests.8d38d0cc
[wasm-tools-net7]
   Installation Source: VS 17.8.34330.188
   Manifest Version:    8.0.0/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.workload.mono.toolchain.net7\8.0.0\WorkloadManifest.json
   Install Type:              Msi

[wasm-tools]
   Installation Source: VS 17.8.34330.188
   Manifest Version:    8.0.0/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.workload.mono.toolchain.current\8.0.0\WorkloadManifest.json
   Install Type:              Msi

Host:
  Version:      8.0.0
  Architecture: x64
  Commit:       5535e31a71

.NET SDKs installed:
  3.1.426 [C:\Program Files\dotnet\sdk]
  6.0.203 [C:\Program Files\dotnet\sdk]
  8.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.27 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.25 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.14 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.26 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.27 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.25 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.14 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.26 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.27 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.25 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.14 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

**Comment by gragra33** [comment]
Have you added logging and set it to debug to capture both your and Microsoft logging, both for WASM and Server, to see where it stops? I would also have the browser console open to check for errors.

**Comment by snapfisher** [comment]
There is no server, and I cannot get it to fail on actual windows...even running in dev tools and set as a low end mobile device.  The console just shows nothing.  There are a bunch of header warnings, but these are all inherent to blazor wasm, not created by me (e.g. "The P3P header should not be used... etc).  There is no console on the mobile browser, right?

**Comment by snapfisher** [comment]
But what I did do was move it back over to .Net 7 and it's now all fixed.   All I did for the move back was a) change the target framework in the project file to net7.0 and 2) move Microsoft.AspNetCore.Components.WebAssembly, Microsoft.AspNetCore.Components.WebAsssembly.DevServer and Microsoft.Authentication.WebAssembly.Msal back from 8.0.0 to 7.0.13.  (I realize that 7.0.14 is the newest version, but this was what I had running before I upgraded).  So, although I cannot put my finger on it, something in the 8.0 packages is just not correct
